Déplacement des extraits (fix #947 et #962)

pierre-24 opened the pull request:
| Q | R |
| --- | --- |
| Correction de bugs ? | oui |
| Nouvelle Fonctionnalité ? | non |
| Tickets concernés | #947, #962 |

Ce qui a été fait :
- Correction de extracts() en `get_extracts()` pour la class Chapter
  (dans /zds/tutorial/models.py)
- Ajout d'une ligne `maj_repo_extract()` manquante dans
  `modify_extract()` et ajout d'un type "move" pour maj_repo_extract()
  (dans /zds/tutorial/views.py)
- Modification de ExtractFactory dans factories.py, qui n'était
  pas complet: ajout de la partie créattion du *.md et commit
  sur repo (dans /zds/tutorial/)
- Ajout d'un PermissionDenied dans la fonction `modify_extract()`
  dans views.py (dans /zds/tutorial/)
- Ajout d'un catch pour gérer les erreurs retourné par `move()`
  dans la fonction modify_extract() dans views.py (dans /zds/
  tutorial/)
- Ajout d'un test unitaire (`test_move_extracts`) dans tests.py
  pour les bigs et mini-tutos + ajout d'extraits à ces TestCase's
  (dans /zds/tutorial/)

Note : à aucun moment le titre n'est modifié, je ne règle pas #913 et #878 par cette PR. Je résous principalement #962 et #947 a déjà été corrigé par @Alex-D au niveau du front.

pierre-24 committed:
Déplacement des extraits (fix #947 et #962)

Ce qui a été fait :
- Correction de extracts() en get_extracts() pour la class Chapter
  (dans /zds/tutorial/models.py)
- Ajout d'une ligne maj_repo_extract() manquante dans
  modify_extract() et ajout d'un type "move" pour maj_repo_extract()
  (dans /zds/tutorial/views.py)

pierre-24 committed:
Déplacement des extraits (fix #947 et #962)

Ce qui a été fait :
- Correction de extracts() en get_extracts() pour la class Chapter
  (dans /zds/tutorial/models.py)
- Ajout d'une ligne maj_repo_extract() manquante dans
  modify_extract() et ajout d'un type "move" pour maj_repo_extract()
  (dans /zds/tutorial/views.py)

artragis commented:
Il serait bien que tu ajoutes un test unitaire qui prouve que ça marche.

artragis commented on zds/tutorial/views.py:
```diff
@@ -1913,15 +1913,15 @@ def modify_extract(request):
              )
         extract.save()
+        maj_repo_extract(request, extract=extract, action="move")
+        
```
pep8 : il te demande d'enlever les espaces sur une ligne vide

pierre-24 replied:
Vu :)

artragis commented on zds/tutorial/views.py:
```diff
@@ -2482,10 +2482,12 @@ def maj_repo_extract(
+            ext.close()
+            index.add([extract.get_path(relative=True)])
+        elif action == "move" :
+            msg = "Déplacement de l'extrait"
```
Le message s'affiche-t-il? car j'ai l'impression que la variable msg est quoi qu'il arrive écrasée par la ligne suivante.

pierre-24 replied:
Bien vu. Ça a été corrigé.

pierre-24 committed:
Fix de quelques erreurs signalées par Atargis

(correction de #947 et #962)

pierre-24 committed:
Fix de quelques erreurs signalées par Atargis

(correction de #947 et #962)

pierre-24 commented:
Je t'avoue que j'ai pas bien compris la mécanique de test et que j'ai peur de faire des erreurs à ce niveau là.

pierre-24 committed:
Corection bug conversion UTF-8

pierre-24 committed:
Corection bug conversion UTF-8

firm1 commented:
> Je t'avoue que j'ai pas bien compris la mécanique de test et que j'ai peur de faire des erreurs à ce niveau là.

Pour le coup, ça va être un peu tendu de faire passer une PR qui modifie cette partie sans test.

C'est vraiment important, ne serait-ce que pour les regressions.

artragis commented:
Ce soir, vient sur le canal smootirc je t'aiderai pour les tests.

pierre-24 commented:
Ok :)

Alex-D commented on zds/tutorial/models.py:
```diff
@@ -614,7 +614,7 @@ def get_absolute_url_online(self):
         return Extract.objects.all().filter(chapter__pk=self.pk).count()
 
-    def extracts(self):
+    def get_extracts(self):
```
Ca m'étonne qu'un changement de nom de méthode ait si peu d'impact sur le code.

Ca veut dire (vu ta PR) que nulle part, la méthod `extracts` n'était appelée ?

Moi je la vois au moins là : https://github.com/zestedesavoir/zds-site/blob/beta-0.2/zds/tutorial/views.py#L1883

pierre-24 replied:
De toute façon, je suis coincé d'ici que j'ai écris le test unitaire. J'en profiterai pour faire le tour du code pour voir l'impact que ça a.

firm1 replied:
Je dirais même qu'après chaque modif du coté des tutos, il faut tester complètement le workflow des tutos en local. Les tests unitaires ne couvrent pas encore tous les cas, donc vaut mieux faire le plus de test possibles.

artragis commented:
Je pense que ça sera une issue à part entière car il faudra lister tout ce qui
- est testable
- n'est pas testé
- peut être rendu testable (revue de code <3)

firm1 commented:
> Je pense que ça sera une issue à part entière car il faudra lister tout ce qui
> - est testable
> - n'est pas testé
> - peut être rendu testable (revue de code <3)

la liste est déjà disponible de manière automatique via [coveralls](https://coveralls.io/r/zestedesavoir/zds-site) visible sur le readme.

Par exemple, si tu veux un aperçu du code qui est testé ou pas actuellement dans le models des tutos, tu peux le voir sur [ce lien](https://coveralls.io/files/215119361). Pour les views.py des tutos, tu l'as [ici](https://coveralls.io/builds/844045), ainsi de suite.

artragis commented:
Coverall te donne les lignes de code atteintes. Aucune sémantique ni vraie analyse des tests. 
Mais si ça vous suffit...

firm1 commented:
> Coverall te donne les lignes de code atteintes. Aucune sémantique ni vraie analyse des tests.
> Mais si ça vous suffit...

Personnellement c'est en regardant les lignes de codes qui ne sont pas testés que je me dis "ah tiens y'a pas de test qui fais ci ou ça". Par exemple sur [ce lien](https://coveralls.io/files/215119361), on voit que les fonctions "get_introduction" et "get_conclusion" ne sont pas touchée par les tests actuellement, ça signifie que soit les méthodes sont inutiles, soit il manque des tests sur des vue offline.

Après, on peut s'amuser à lister ça proprement dans une issue, et la maintennir à jour, pourquoi pas. Mais je trouverai ça redondant.

pierre-24 committed:
Intégration d'un test unitaire (fix #947 et #962)

Ce qui a été fait :
- Modification de ExtractFactory dans factories.py, qui n'était
  pas complet: ajout de la partie créattion du *.md et commit
  sur repo (dans /zds/tutorial/)
- Ajout d'un PermissionDenied dans la fonction modify_extract()
  dans views.py (dans /zds/tutorial/)
- Ajout d'un catch pour gérer les erreurs retourné par move()
  dans la fonction modify_extract() dans views.py (dans /zds/
  tutorial/)
- Ajout d'un test unitaire (test_move_extracts) dans tests.py
  pour les bigs et mini-tutos + ajout d'extraits à ces TestCase's
  (dans /zds/tutorial/)

ATTENTION : possible recouvrement avec #969 à vérifier pour le
remplacement de extracts() en get_extracts()

pierre-24 committed:
Intégration d'un test unitaire (fix #947 et #962)

Ce qui a été fait :
- Modification de ExtractFactory dans factories.py, qui n'était
  pas complet: ajout de la partie créattion du *.md et commit
  sur repo (dans /zds/tutorial/)
- Ajout d'un PermissionDenied dans la fonction modify_extract()
  dans views.py (dans /zds/tutorial/)
- Ajout d'un catch pour gérer les erreurs retourné par move()
  dans la fonction modify_extract() dans views.py (dans /zds/
  tutorial/)
- Ajout d'un test unitaire (test_move_extracts) dans tests.py
  pour les bigs et mini-tutos + ajout d'extraits à ces TestCase's
  (dans /zds/tutorial/)
- rebase (!)

ATTENTION : possible recouvrement avec #969 à vérifier pour le
remplacement de extracts() en get_extracts()

pierre-24 committed:
pep8

pierre-24 committed:
pep8

sandhose committed:
Merge pull request #984 from pierre-24/fix_926

fix #926 (tutoriels en béta)

pierre-24 committed:
Merge branch 'tentative_947_962' of https://github.com/pierre-24/zds-site into tentative_947_962

Conflicts:
	zds/tutorial/tests.py

pierre-24 commented:
**Pour les reviews et autres :** j'attend que la PR #971  soit passée pour éventuellement penser à merger celle-ci. D'ici là, y'aura surement un _rebase_ et un _squash_ à faire.

ArnaudCalmettes commented:
Comme vu avec artragis (devzone + IRC), le problème adressé par cette PR est également géré par la #1001 .
Du coup, on peut fermer celle-ci.
